compiler_opt/rl/log_reader.py

```diff
@@ -120,6 +127,14 @@ def __init__(self, spec: tf.TensorSpec, buffer: bytes):
   def spec(self):
     return self._spec
 
+  @property
+  def buffer(self):
```

Comment:
why do you need this and the `len` property? should they have a small unit test, too? 

also, maybe `buffer` -> `raw_bytes` to make it clear it's that, not another way to dereference the typed view

and then `len` -> do you need to use it as a length of the raw buffer? if so, maybe call it like that and return the _len multiplied by the scalar type size.

Reply:
I convert the data given by the log_reader to a numpy array, and the easiest way to do that was to use https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html.

I use numpy in the environments instead of tensorflow because numpy is universally used in every ML framework (tf, pytorch, jax) and I want to start decoupling things from TF when they don't fundamentally require it.

I'm not sure if they need unit tests - if you want I can add them, but they're not doing anything surprising. I think if the other tests fail it should catch any issues here.

Reply:
But the object itself acts as a buffer, it has an indexer and a __len__.

Reply:
np has its own notion of a buffer interface which is pretty specific, and if we naively try to pass one of these objects to `np.frombuffer` we get the error:
```
 TypeError: a bytes-like object is required, not 'LogReaderTensorValue'
```

Reply:
ha! interesting. OK. so can we do with just a `raw_bytes`, looks like len() wouldn't be needed?

Reply:
ok, replaced `raw_bytes` and `len` with a `to_numpy` method directly, and added a unit test for it.

Reply:
a, neat. Thanks!